_core/smart/overview.md

```diff
@@ -0,0 +1,135 @@
+</p>
+</div>
+
+## Who should use smart contracts?
+
+You can use Clarity to write standalone contracts or to write contracts that are part of decentralized applications (DApps) you write with the blockstack.js library. Smart contracts allow two parties to exchange anything of value (money, property, shares), in an automated, auditable, and secure way _without the services of a middleman_. Nick Szabo introduced the canonical metaphor for smart contracts, a vending machine. 
+
+In Nick Szabo's metaphor, the vending machine is the smart contract. The buyer and machine owner are the two parties. A vending machine executes a set of hard-coded actions when the buyer engages with it. The machine displays the items and their prices. A buyer enters money into the machine which determines if the amount fails to mee, meets, or exceeds an item's price. Based on the amount, the machine asks for more money, dispenses an item, or dispenses and item and change. 
+
+Not every application requires smart contracts. If you are not sure or are new to smart contracts concepts, you should read <a href="https://blockgeeks.com/guides/smart-contracts/" target="_blank">a good general explanation of smart contracts</a> before working with Clarity. 
+
+## Language and program design
+
+Clarity differs from most other smart contract languages in two essential ways:
+
+* The language is not intended to be compiled.
+* The language is not Turing complete. 
+  
+These differences allow for static analysis of programs to determine properties like runtime cost and data usage. 
+
+A Clarity smart contract is composed of two parts &mdash; a data-space and a set of functions.  Only the associated smart contract may modify its corresponding data-space on the blockchain. Functions are private unless they are defined as public functions. Users call smart contracts' public functions by broadcasting a transaction on the blockchain which invokes the public function.
+
+Contracts can also call public functions from other smart contracts. The ability to do a static analysis of a smart contract allows a user to determine dependency between contracts.
+
+## The coding environment
+
+Clarity is a list processing (LISP) language, as such it is not compiled. Omitting compilation prevents the possibility of error or bugs introduced at the compiler level. You can write Clarity smart contract programs on any operating system with a text editor. You can use any editor you are comfortable with such as Atom, Vim, or even Notepad. The Clarity files you create with an editor have a `.clar` extension.
+
+Clarity is in pre-release and does not yet directly interact with the live Stacks blockchain. For the pre-release period you need a virtual machine (VM) environment to run Clarity contracts. Blockstack provides a Docker VM that you can use or you can build a test environment locally from code. Either the VM or a local environment is sufficient for testing Clarity programming for standalone contracts.
```

Comment:
> you need a virtual machine (VM) environment

The use of "virtual machine" here could be a little confusing -- users just need to setup a local environment, not necessarily with a virtual machine or docker image.

Reply:
I want to say "test environment" which is either the VM or a build in their local environment. Does that work?

Reply:
Yep that works

Reply:
Cool. Changed through out the doc.